### ISSUE TITLE
Huge performance difference between CPU and GPU

### ISSUE THREAD

**Issue by shaanchandra** [pyg-team/pytorch_geometric#1240]
## 🐛 Bug

When I trained the models on my local CPU, everything worked as I would expect. However, when I try to train on GPU the models break. More specifically, the F1 for a binary classification task remains 0 throughout training on GPU but on CPU (with exact same settings, seeds, etc) achieves ~80 F1. The architecture is a simple 2 layer GNN (I have tried GCN and GAT and both have the same behavior).

If you look at the sample outputs below from both the devices you can see that the losses on GPU are also higher. It looked like the gradients were there for just the classification head (MLP). But I printed gradients for both the cases and there are gradients for all the components of the model in each case (2 layers of GCN/GAT + classification layer). 

Any pointers what might be going wrong on the GPU?
Am I missing something? Like some setting/parameter to set? 

## Expected behavior

Output in case of GPU:

```
Epoch: 1/100,            
train_loss = 0.5891,    train_Acc = 0.7685,    train_prec = 0.0000,    train_recall = 0.0000,    train_f1 = 0.0000,            
eval_loss = 0.5305,     eval_acc = 0.7682,     eval_prec = 0.0000,     eval_recall = 0.0000,     eval_f1 = 0.0000                
lr  =  0.00100000
Elapsed Time:  00:00:11.53
1

Epoch: 2/100,            
train_loss = 0.5694,    train_Acc = 0.7685,    train_prec = 0.0000,    train_recall = 0.0000,    train_f1 = 0.0000,            
eval_loss = 0.5497,     eval_acc = 0.7682,     eval_prec = 0.0000,     eval_recall = 0.0000,     eval_f1 = 0.0000                
lr  =  0.00100000
Elapsed Time:  00:00:22.02
2

Epoch: 3/100,            
train_loss = 0.5568,    train_Acc = 0.7685,    train_prec = 0.0000,    train_recall = 0.0000,    train_f1 = 0.0000,            
eval_loss = 0.5047,     eval_acc = 0.7682,     eval_prec = 0.0000,     eval_recall = 0.0000,     eval_f1 = 0.0000                
lr  =  0.00100000
Elapsed Time:  00:00:32.58
3

Epoch: 4/100,            
train_loss = 0.5542,    train_Acc = 0.7685,    train_prec = 0.0000,    train_recall = 0.0000,    train_f1 = 0.0000,            
eval_loss = 0.5315,     eval_acc = 0.7682,     eval_prec = 0.0000,     eval_recall = 0.0000,     eval_f1 = 0.0000                
lr  =  0.00064000
Elapsed Time:  00:00:43.12
4

Epoch: 5/100,            
train_loss = 0.5502,    train_Acc = 0.7685,    train_prec = 0.0000,    train_recall = 0.0000,    train_f1 = 0.0000,            
eval_loss = 0.5302,     eval_acc = 0.7682,     eval_prec = 0.0000,     eval_recall = 0.0000,     eval_f1 = 0.0000                
lr  =  0.00080000
Elapsed Time:  00:00:53.59
5
```

Output in case of CPU:

```
Epoch: 1/100,            
train_loss = 0.3761,    train_Acc = 0.8277,    train_prec = 0.6361,    train_recall = 0.5973,    train_f1 = 0.6161,            
eval_loss = 0.2803,     eval_acc = 0.8956,     eval_prec = 0.7908,     eval_recall = 0.7469,     eval_f1 = 0.7683                
lr  =  0.00100000
Elapsed Time:  00:00:57.67
New High Score! Saving model...

Epoch: 2/100,            
train_loss = 0.3129,    train_Acc = 0.9042,    train_prec = 0.8332,    train_recall = 0.7328,    train_f1 = 0.7798,            
eval_loss = 0.2795,     eval_acc = 0.8875,     eval_prec = 0.7395,     eval_recall = 0.7942,     eval_f1 = 0.7659                
lr  =  0.00100000
Elapsed Time:  00:02:01.88
1

Epoch: 3/100,            
train_loss = 0.2775,    train_Acc = 0.9197,    train_prec = 0.8476,    train_recall = 0.7962,    train_f1 = 0.8211,            
eval_loss = 0.2673,     eval_acc = 0.9008,     eval_prec = 0.8131,     eval_recall = 0.7428,     eval_f1 = 0.7763                
lr  =  0.00100000
Elapsed Time:  00:03:17.85
New High Score! Saving model...

Epoch: 4/100,            
train_loss = 0.2518,    train_Acc = 0.9326,    train_prec = 0.8734,    train_recall = 0.8292,    train_f1 = 0.8507,            
eval_loss = 0.3062,     eval_acc = 0.8774,     eval_prec = 0.6978,     eval_recall = 0.8313,     eval_f1 = 0.7587                
lr  =  0.00064000
Elapsed Time:  00:04:28.53
1

Epoch: 5/100,            
train_loss = 0.2350,    train_Acc = 0.9318,    train_prec = 0.8767,    train_recall = 0.8209,    train_f1 = 0.8479,            
eval_loss = 0.2778,     eval_acc = 0.9046,     eval_prec = 0.8373,     eval_recall = 0.7305,     eval_f1 = 0.7802                
lr  =  0.00080000
Elapsed Time:  00:05:42.11
New High Score! Saving model...
```

## Environment

**CPU device details:**
* OS: Windows
* Python version: 
* PyTorch version: 1.4
* CUDA/cuDNN version: NA (CPU)
* GCC version: NA
* Any other relevant information:

**GPU device details:**

* OS: Linux
* Python version: 3.6.3
* PyTorch version: 1.5
* CUDA/cuDNN version: 10.1 / 7.6.5
* GCC version: 
* Any other relevant information:

## Additional context
I have made sure that torch-CUDA and device CUDA versions are same (`10.1` and `10.1.243` respectively). I hope that works?
For the CPU device I have installed `cpu` versions of the libraries and for GPU device I have installed `cu101` versions

**Comment by shaanchandra** [comment]
UPDATE:
Works on GPU correctly for Torch 1.4 and CUDA 10.1
So I guess Torch 1.5 is causing issues (even though did not throw any warnings or errors)

**Comment by rusty1s** [comment]
Can you confirm a valid `torch-scatter` installation on PyTorch 1.5 by running the test suite of `torch-scatter`? That would be great to identify the error.
